src/gui/search/searchwidget.cpp

```diff
@@ -132,10 +141,29 @@ namespace
         return tabName;
     }
 
+    nonstd::expected<QStringList, QString> loadHistory(const Path &filePath, const int limit)
+    {
+        const auto readResult = Utils::IO::readFile(filePath, HISTORY_FILE_MAX_SIZE);
+        if (!readResult)
+        {
+            if (readResult.error().status == Utils::IO::ReadError::NotExist)
+                return {};
+
+            return nonstd::make_unexpected(readResult.error().message);
+        }
+
+        QByteArray historyData = readResult.value();
+        QStringList history;
+        QTextStream stream {&historyData};
```

Comment:
You could just call `split()` on `historyData` instead. Just a suggestion, I don't insist.

Reply:
Or do you mean `QString(historyData).split(u'\n')`? But it creates one extra `QString` containing all the input data.

Reply:
>It returns QList<QByteArray> so I still need to convert each item to QString. What's the benefit?

The `QTextStream` is horribly slow the last time I tested. For small code, the inconvenience of an additional conversion to `QString` is minimal IMO. You can even use `Utils::ByteArray::splitToViews(Qt::SkipEmptyParts)`.

Reply:
> The `QTextStream` is horribly slow the last time I tested.

Well, if that's the case, then it makes sense.


> For small code, the inconvenience of an additional conversion to `QString` is minimal IMO.

It's not about the code, but about the amount of data that will be duplicated in memory. But I hope it's not critical in this case. Besides, `Utils::ByteArray::splitToViews` seems to help avoid this duplication.